src/ussd_popi_user_data.js

```diff
@@ -950,25 +953,44 @@ go.app = function() {
       });
 
       self.add('state_new_number_channel', function(name) {
-          return new ChoiceState(name, {
-              question: $('Thank you. How would you like to receive messages about you and your baby?'),
-              choices: [
-                  new Choice('whatsapp', $('WhatsApp')),
-                  new Choice('sms', $('SMS'))
-              ],
-              next: 'state_switch_number'
-          });
+          var msisdn = utils.normalize_msisdn(self.im.user.get_answer('state_enter_new_phone_number'), "27");
+          var channel;
+          return self.is_valid_recipient_for_pilot({ // msisdn whatsappable
+            address: msisdn,
+            wait: true,
+        }).then(function(confirmed) {
+            if(confirmed) {
+                channel = 'whatsapp';
+                self.map_channel(channel);
```

Comment:
`map_channel` just takes the channel type, and maps it to a user readable format. Since we're not doing anything with the result here, we can just remove the two `map_channel` calls.